src/circuit/circuit_frame.rs

```diff
@@ -3055,6 +3151,168 @@ fn apply_continuation<F: LurkField, CS: ConstraintSystem<F>>(
 
         let res = AllocatedPtr::from_parts(&res_tag, &val);
 
+        // Next constraints are used for number comparisons: <, <=, >, >=
+        ///////////////////////////////////////////////////////////////////////
+        let double_a = constraints::add(&mut cs.namespace(|| "double a"), a, a)?;
+        // Ideally we would compute the bit decomposition for a, not for 2a,
+        // since it would be possible to use it for future purposes.
+        // TODO: replace bit decompositon from 2a to a.
+        let double_a_bits = double_a
+            .to_bits_le_strict(&mut cs.namespace(|| "double a lsb"))
+            .unwrap();
+        let lsb_2a = double_a_bits.get(0);
+
+        let double_b = constraints::add(&mut cs.namespace(|| "double b"), b, b)?;
+        // TODO: replace bit decompositon from 2b to b.
+        let double_b_bits = double_b
+            .to_bits_le_strict(&mut cs.namespace(|| "double b lsb"))
+            .unwrap();
+        let lsb_2b = double_b_bits.get(0);
+
+        let diff_is_zero = alloc_is_zero(&mut cs.namespace(|| "diff is zero"), &diff)?;
+        let double_diff = constraints::add(&mut cs.namespace(|| "double diff"), &diff, &diff)?;
+        // TODO: replace bit decompositon from 2diff to diff.
+        let double_diff_bits = double_diff.to_bits_le_strict(&mut cs).unwrap();
+        let lsb_2diff = double_diff_bits.get(0);
+
+        // We have that a number is defined to be negative if the parity bit (the
+        // least significant bit) is odd after doubling, meaning that the field element
+        // (after doubling) is larger than the underlying prime p that defines the
+        // field, then a modular reduction must be carried out, changing the parity that
```

Comment:
I think you mean 'must have been carried out' (i.e. that is how the parity 'changed' from expected), rather than 'must be carried out' (which reads to me as suggesting that some further reduction is required).